packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -617,18 +664,13 @@ class ContainerLayer extends Layer {
   }
 
   @override
-  Iterable<S> findAll<S>(Offset regionOffset) {
-    Iterable<S> result = Iterable<S>.empty();
-    if (firstChild == null)
-      return result;
-    Layer child = lastChild;
-    while (true) {
-      result = result.followedBy(child.findAll<S>(regionOffset));
-      if (child == firstChild)
-        break;
-      child = child.previousSibling;
+  bool hitTest<S>(LayerHitTestResult<S> result, {Offset regionOffset}) {
+    for (Layer child = lastChild; child != null; child = child.previousSibling) {
+      final bool isAbsorbed = child.hitTest<S>(result, regionOffset: regionOffset);
+      if (isAbsorbed)
+        return true;
     }
-    return result;
+    return false;
```

Comment:
do we intentionally not add this layer to the list as well?

Reply:
As far as it's designed right now, layer hit test only works for annotations. Therefore only `AnnotatedRegionLayer` can add anything to the list.
Also it is not the layers that are added, but the value.

Reply:
seems weird to have hit testing work so differently in layers than in render objects.

Reply:
Honestly I was about to name it `findAnnotations` since their implementations are so different, but I couldn't think of a shorter name.